### PR TITLE
test: add unit testing

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Split up in a separate PR to reduce noise and focus on changes if needed.

### CODE REVIEW

**Review by OrKoN — APPROVED** [review]

File: src/bidiMapper/domains/network/NetworkStorage.spec.ts [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+
+      const event = await getEvent('network.beforeRequestSent');
+      expect(event).to.exist;
+    });
+
+    it('should work for reverse order', async () => {
+      const request = new MockCdpNetworkEvents(cdpClient);
+
+      request.requestWillBeSend();
+      request.requestWillBeSendExtraInfo();
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkStorage.spec.ts:347]: order is not reversed

File: src/bidiMapper/domains/network/NetworkStorage.spec.ts [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    } as unknown as CdpClient);
+
+    networkStorage.onCdpTargetCreated(cdpTarget);
+  });
+
+  describe('network.beforeRequestSent', () => {
+    it('should work for normal order', async () => {
+      const request = new MockCdpNetworkEvents(cdpClient);
+
+      request.requestWillBeSend();
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkStorage.spec.ts:336]:
```suggestion
      request.requestWillBeSent();
```

Comment (OrKoN) [reply src/bidiMapper/domains/network/NetworkStorage.spec.ts:336]: in all helpers

File: src/bidiMapper/domains/network/NetworkStorage.spec.ts [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+      event = await getEvent('network.responseStarted');
+
+      expect(event).to.deep.include({
+        isBlocked: true,
+        intercepts: [interception],
+      });
+    });
+
+    it('should work non blocking interception ', async () => {
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkStorage.spec.ts:469]:
```suggestion
    it('should work non blocking interception', async () => {
```

File: src/bidiMapper/domains/network/NetworkStorage.spec.ts [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+
+      let event = await getEvent('network.responseStarted');
+      expect(event).to.not.exist;
+
+      request.responsePaused();
+      event = await getEvent('network.responseStarted');
+      expect(event).to.not.exist;
+
+      request.responseReceived();
+      request.responseReceivedExtraInfo();
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkStorage.spec.ts:487]: For responseReceivedExtraInfo we should also have variations with the flags indicating the extra info

File: src/bidiMapper/domains/network/NetworkStorage.spec.ts [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+      on(): void {
+        // Used for clearing on target disconnect
+      },
+    } as unknown as CdpClient);
+
+    networkStorage.onCdpTargetCreated(cdpTarget);
+  });
+
+  describe('network.beforeRequestSent', () => {
+    it('should work for normal order', async () => {
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkStorage.spec.ts:333]: also need a test for redirects

**Comment by Lightning00Blade** [comment]
Too hard to fix extracting the change into a new PR.

**Pull request closed without merging**